### DISCUSSION TITLE
Equation based surface creation using PyAEDT

### DISCUSSION THREAD

**Discussion by Abhi-SimuTech** [ansys/pyaedt#5079] [Q&A] [answered]
Hi,

I wish to create arbitrary equation-based surfaces in HFSS using PyAEDT &/or Python. 

HFSS does have "draw equation based surface" feature, but I was hoping that with the link with python we may have more flexibilities creating complex surfaces in PyAEDT environment.

However, I was unable to find a method that says something like create equation based surface when I searched for it. I did find create_equationbased_curve like you can see below:

![image](https://github.com/user-attachments/assets/1b3601ec-5244-4920-adfb-cfe130f8a06e)

Could you suggest what would be the best way to draw equation based surfaces using PyAEDT? It will be great if you could point me to an example. 

P.S. I remember seeing a rough surface (not sure if that was for creating random sea surface fluctuations or for other purpose) creation workflow using scripting and SpaceClaim/HFSS that I can not find it now.)

Regards
Abhi

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @Abhi-SimuTech ,

You can use the native equation based surface:

https://aedt.docs.pyansys.com/version/stable/API/_autosummary/ansys.aedt.core.modeler.modeler_3d.Modeler3D.create_equationbased_surface.html

It has an example.

But, if you find limitations on the implemented AEDT modeler, you will have the same limitations here because the PyAEDT method is using it. 

If you need more complex surfaces, then as you said, you can create it in SpaceClaim, and then import the SpaceClaim geometry using this method:
https://aedt.docs.pyansys.com/version/stable/API/_autosummary/ansys.aedt.core.modeler.modeler_3d.Modeler3D.import_spaceclaim_document.html
